File: autovalue-example/src/main/java/nz/bradcampbell/autovalueexample/State.java
package nz.bradcampbell.autovalueexample;

import com.google.auto.value.AutoValue;
import android.os.Parcelable;
import android.support.annotation.Nullable;

import java.util.Date;

@AutoValue
public abstract class State implements Parcelable {
    public abstract int count();
    public abstract Date modificationDate();
    @Nullable public abstract String nullableString();

    public static State create(int count, Date modificationDate, String nullableString) {
        return new AutoValue_State(count, modificationDate, nullableString);
    }
}

File: kotlin-example/src/main/kotlin/nz/bradcampbell/kraftpaper/kotlinexample/State.kt
package nz.bradcampbell.kraftpaper.kotlinexample

import nz.bradcampbell.kraftpaper.KraftPaper
import java.util.*

@KraftPaper(typeAdapters = arrayOf(DateTypeAdapter::class))
data class State(val count: Int, val modificationDate: Date)
